code/network/multiutil.cpp

```diff
@@ -1910,19 +1913,24 @@ int multi_can_end_mission(net_player *p)
 
 	// only wing/team leaders can end the mission
 	case MSO_END_LEADER:
+	{
 		// if the player has an invalid object #
 		if(p->m_player->objnum < 0){
 			return 0;
 		}
 
 		// check to see if he's a wingleader
-		sp = &Ships[Objects[p->m_player->objnum].instance];
-		if (sp->ship_name[strlen(sp->ship_name)-1] == '1') 
+		const ship_registry_entry* ship_regp = ship_registry_get(Ships[p->m_player->objnum].ship_name);
+		if (ship_regp == nullptr){
```

Comment:
Ditto on the Assertion

Reply:
Good to go?

Reply:
See above comment about the if() not being needed.